File: frontend/src/api/fire.ts
import type { FIReData, CreditSimulationResult } from '../types/fire';
import { apiFetch } from './client';

export function fetchFIReData(): Promise<FIReData> {
  return apiFetch<FIReData>('/api/fire/relationships');
}

export function simulateCredit(cardId: string, passed: boolean): Promise<CreditSimulationResult> {
  return apiFetch<CreditSimulationResult>('/api/fire/simulate', {
    method: 'POST',
    body: JSON.stringify({ card_id: cardId, passed }),
  });
}

File: backend/app/routers/fire.py
"""FIRe API endpoints."""

from fastapi import APIRouter

from app.models.fire import CreditSimRequest, CreditSimResult, FIReData
from app.services import fire_service

router = APIRouter(prefix="/api/fire", tags=["fire"])


@router.get("/relationships", response_model=FIReData)
def get_relationships():
    """Get all encompassing relationships."""
    return fire_service.get_fire_data()


@router.post("/simulate", response_model=CreditSimResult)
def simulate_credit(req: CreditSimRequest):
    """Simulate credit flow for a card."""
    return fire_service.simulate_credit(req.card_id, req.passed)


@router.get("/heatmap")
def get_heatmap():
    """Get heatmap data for visualization."""
    return fire_service.get_heatmap_data()
